plasmapy/particles/particle_class.py

```diff
@@ -96,6 +104,91 @@ def _category_errmsg(particle, category: str) -> str:
     return errmsg
 
 
+class ParticleJSONDecoder(json.JSONDecoder):
+    """
+    A custom `JSONDecoder` class to deserialize particles.
+    Inherits from `json.JSONDecoder`. Refer to documentation at:
+    https://docs.python.org/3/library/json.html#json.JSONDecoder
+
+    Parameters
+    ----------
+    object_hook
+        operation to be performed to deserialize the
+        JSON string to a `dict`. The ParticleJSONDecoder uses a custom hook
+        called `particle_hook` to deserialize particle JSON strings.
+
+    parse_float
+        Specifies how to handle floating point numbers
+        in a JSON string. Refer to documentation link above for more details.
+
+    parse_int
+        Specifies how to handle integer numbers.
+        Refer to documentation link above for more details.
+
+    parse_constant
+        Specifies how to handle constants.
+        Refer to documentation link above for more details.
+
+    strict
+        Refer to documentation link above for details.
+
+    object_pairs_hook
+        Refer to documentation link above for details.
+    """
+
+    def __init__(
+        self,
+        *,
+        object_hook=None,
+        parse_float=None,
+        parse_int=None,
+        parse_constant=None,
+        strict=True,
+        object_pairs_hook=None,
+    ):
+
+        if object_hook is None:
+            object_hook = self.particle_hook
+
+        json.JSONDecoder.__init__(
+            self,
+            object_hook=object_hook,
+            parse_float=parse_float,
+            parse_int=parse_int,
+            parse_constant=parse_constant,
+            strict=strict,
+            object_pairs_hook=object_pairs_hook,
+        )
```

Comment:
```suggestion
    """
    A custom `~json.JSONDecoder` class to deserialize JSON objects into
    PlasmaPy particle objects.

    Parameters
    ----------
    object_hook:
        If specified, will be called with the result of every JSON object
        decoded and its return value will be used in place of the given `dict`.
        This can be used to provide custom deserializations (e.g. to support
        JSON-RPC class hinting)  (If ot specified, then defaults
        `particle_hook`.)

    **kwargs:
        Any keyword accepted by `~json.JSONDecoder`.
    """

    def __init__(self, *, object_hook=None, **kwargs):
        if object_hook is None:
            object_hook = self.particle_hook

        json.JSONDecoder.__init__(self, object_hook=object_hook, **kwargs)
```

I know I suggested a lot of the original code for this, and I apologize changing this up on you.  I think we should only document what we're doing differently from `JSONDecoder` and heavily rely on `JSONDecoder`'s own documentation.  No reason to reinvent the wheel.

Reply:
No worries! I agree that this will make it easier to understand for future reference as well. Thanks for the suggestion :)